tutorials/singularity-containers-with-cloud-build/README.md

```diff
@@ -5,152 +5,309 @@ author: wardharold
+
 ```bash
-source ./env.sh
+
+# List organizations - you can use the name or the id
+$ gcloud organizations list
+
+# List billing accounts
+$ gcloud alpha billing accounts list
+
+# List projects
+$ gcloud projects list
+
+# List zones
+$ gcloud compute zones list
+
 ```
 
-### Create your project
+When you have your env.sh template filled in, source it! This will 
+define the environment variables to your current shell session.
+
 ```bash
-gcloud projects create $PROJECT --organization=$ORG
+$ source ./env.sh
 ```
+
+
+### Step 2: (Optional) Project Creation
+
+If you defined a new project in the step above, you need to use the gcloud
+command to create it. Otherwise, skip over this step. Nothing to see here, folks.
+
 ```bash
-gcloud beta billing projects link $PROJECT --billing-account=$(gcloud beta billing accounts list | grep $BILLING_ACCOUNT | awk '{print $1}')
+$ gcloud projects create $PROJECT --organization=$ORG
 ```
 ```bash
-gcloud config configurations create -- activate $PROJECT
+$ gcloud beta billing projects link $PROJECT --billing-account=$(gcloud beta billing accounts list | grep $BILLING_ACCOUNT | awk '{print $1}')
 ```
 ```bash
-gcloud config set compute/zone $ZONE
+$ gcloud config configurations create -- activate $PROJECT
 ```
 
-## Enable the required Google APIs
-```bash
-gcloud services enable compute.googleapis.com
-```
+### Step 3: Enable APIs and Set Project
+
+Once you've created the project, it's good to ensure it's the active one.
+
 ```bash
-gcloud services enable cloudbuild.googleapis.com
+$ gcloud config set compute/zone $ZONE
+Updated property [compute/zone].
 ```
+
+The following commands will enable Google Compute Engine, Google Cloud Builder,
+and Google Container Registry:
+
 ```bash
-gcloud services enable containerregistry.googleapis.com
+$ gcloud services enable compute.googleapis.com
+$ gcloud services enable cloudbuild.googleapis.com
+$ gcloud services enable containerregistry.googleapis.com
 ```
 
-## Create a custom Singularity build step
+## Singularity Build
+
+The next section will walk you through issuing a gcloud command to build
+a container. Specifically, the first step is *not* actually building the container 
+- it's basically defining the builder we will use in future steps (with build recipes).
+Once we have this custom build step defined, then we can use it against a Singularity recipe to build a container and send it to Google Storage. In summary:
+
+ - builder.yaml will create our builder
+ - cloudbuild.yaml will run a cloud build
+
+Let's go!
+
+### Step 1. Create a Singularity builder
 
-Cloud Build supports the definition and use of *custom build steps* to extend the range of tasks it can handle.
-You will create a Singularity custom build step to use when building your Singularity containers. 
+Cloud Build supports the definition and use of *custom build steps* to extend the range of tasks it can handle. We create our builder via a custom build step, and can
```

Comment:
Cloud Build supports the definition and use of *custom build steps* to extend the range of tasks it can handle. ~~We~~You create ~~our~~the builder via a custom build step, and can